src/devices/src/display/gtk/worker.rs

```diff
@@ -0,0 +1,278 @@
+use std::os::fd::AsRawFd;
+
+use crate::display::gtk::DisplayEvent;
+use crate::display::{DisplayInfo, DisplayInfoList, MAX_DISPLAYS};
+use utils::pollable_channel::PollableChannelReciever;
+
+use glib::{source, ControlFlow, IOCondition};
+use gtk4::{
+    gdk,
+    gio::{ActionEntry, Cancellable, SimpleActionGroup},
+    prelude::{ActionMapExtManual, GtkWindowExt, WidgetExt},
+    AlertDialog, Align, Button, EventControllerMotion, HeaderBar, Overlay, Picture, Revealer,
+    Window,
+};
+
+struct Scanout {
+    window: Window,
+    width: i32,
+    height: i32,
+    format: gdk::MemoryFormat,
+    picture: Picture,
+}
+
+impl Scanout {
+    fn new(
+        title: String,
+        display_info: DisplayInfo,
+        width: i32,
+        height: i32,
+        format: gdk::MemoryFormat,
+    ) -> Self {
+        let header_bar = HeaderBar::builder().build();
+        let window = Window::builder()
+            .title(title)
+            // remove the close button, since it is unclear what it should do
+            .deletable(false)
+            // Enforce a minimum window size:
+            .height_request(64)
+            .titlebar(&header_bar)
+            .visible(true)
+            .build();
+
+        let actions = SimpleActionGroup::new();
+        actions.add_action_entries([
```

Comment:
why use a separate action group ? i would recommend using a gtk::Window which is an gio::ActionMap already